docs: Added documentation about "AlternateContent" feature (#841)
unoplatform/Uno.Themes: doc/material-controls-extensions.md

# Material Control Extensions

## Icon

> This feature allows to put an icon on some supported control. Those icons could be any kind of _Icon_: `<BitmapIcon />`, `<FontIcon />`, `<SymbolIcon/>`, or `<PathIcon />`.

Reviewer: ```suggestion
This feature allows putting an icon on some supported control. Those icons could be any kind of _Icon_: `<BitmapIcon />`, `<FontIcon />`, `<SymbolIcon/>`, or `<PathIcon />`.

```

> This feature allows to put an icon on some supported control. Those icons could be any kind of _Icon_: `<BitmapIcon />`, `<FontIcon />`, `<SymbolIcon/>`, or `<PathIcon />`.

Reviewer 2: ```suggestion
This feature allows for the addition of icon on the supported controls. Those icons could be any of the [`IconElement`](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.iconelement)s: `<BitmapIcon />`, `<FontIcon />`, `<PathIcon />` or `<SymbolIcon />`.
```

Reviewer 2: might want to clarify which icons are supported and on what platforms here
i have a feeling that we probably dont support all of them..?
perhaps we can expand the [IconElement doc](https://platform.uno/docs/articles/implemented/windows-ui-xaml-controls-iconelement.html) and link it here? or, it is already somewhere?

Here are supported control with samples:
* TextBox:
  ``` xml
    <TextBox Style="{StaticResource MaterialFilledTextBoxStyle}">
        <um:ControlExtensions.Icon>
            <SymbolIcon Symbol="SolidStar" />
        </um:ControlExtensions.Icon>
    </ComboBox>
  ```
* ComboBox:
  ``` xml
    <ComboBox Style="{StaticResource MaterialComboBoxStyle}">
        <um:ControlExtensions.Icon>
            <SymbolIcon Symbol="SolidStar" />
        </um:ControlExtensions.Icon>
    </ComboBox>
  ```

## Alternate Content

> This feature allows to put a different content on a control when the state changes.

Reviewer 3: What's the state property? Is it per control?

Reviewer 2: control specific, and only currently ToggleButton uses it.

> This feature allows to put a different content on a control when the state changes.

Reviewer: ```suggestion
This feature allows putting different content on a control when the state changes.
It's control specific and for now, you can only use it with the ToggleButton control.
```

### Alternate Content on ToggleButton
``` xml
    <ToggleButton Style="{StaticResource MaterialToggleButtonIconStyle}">
        <!-- This is the default content - which is when the control state is UNCHECKED (the default value of a ToggleButton) -->
        <PathIcon Data="{StaticResource Icon_more_horizontal}" />

>         <um:ControlExtensions.AlternateContent>
>         <!-- This is the alternate content - which is when the control state is CHECKED -->

Reviewer 2: indent alignment:
```suggestion
        <!-- This is the alternate content - which is when the control state is CHECKED -->
        <um:ControlExtensions.AlternateContent>
```
